Shared/Samples/Densify and generalize geometry/DensifyAndGeneralizeGeometryView.swift

```diff
@@ -0,0 +1,184 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct DensifyAndGeneralizeGeometryView: View {
+    /// The view model for the sample.
+    @StateObject private var model = Model()
+    
+    /// A Boolean value indicate whether the option sheet is showing.
+    @State private var isShowingOptions = false
+    
+    var body: some View {
+        // Create a map view to display the map.
+        MapView(map: model.map, graphicsOverlays: [model.graphicsOverlay])
+            .toolbar {
+                ToolbarItem(placement: .bottomBar) {
+                    Button("Options") {
+                        isShowingOptions = true
+                    }
+                    .sheet(isPresented: $isShowingOptions, detents: [.medium], dragIndicatorVisibility: .visible) {
+                        OptionsView(model: model)
+                    }
+                }
+            }
+    }
+}
+
+extension DensifyAndGeneralizeGeometryView {
+    /// The view model for the sample.
+    class Model: ObservableObject {
+        /// A map with a Streets (Night) basemap centered on the polyline.
+        lazy var map: Map = {
+            let map = Map(basemapStyle: .arcGISStreetsNight)
+            
+            // Set the initial viewpoint to show the extent of the graphics.
+            map.initialViewpoint = Viewpoint(
+                center: originalPolyline.extent.center,
+                scale: 65907)
+            return map
+        }()
```

Comment:
A general rule of thumb of mine, if a property is guaranteed to be used, don't use `lazy` to manipulate its initialization order. lazy should be conservatively used for those resource consuming types that might not be used (FileManager as an example). There should be a way to play with the order to make everything load.